### PR TITLE
Stabilize two `flake8-pyi` rules

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

Stabilize `PYI057` and `PYI62`.

The rules have been added more than 90 days ago and there are no outstanding issues (searching by rule name or code). 

It is a bit funny that the rules apply to both `pyi` and `py` files, considering that they are part of `flake8-pyi`.
They do match the behavior of the upstream rules and we can move the rules into a more appropriate group when doing the rule categorization.

## Test Plan

See ecosystem check

### CODE REVIEW

File: crates/ruff_linter/src/codes.rs [hunk 1]
```diff
@@ -779,9 +779,9 @@ pub fn code_to_rule(linter: Linter, code: &str) -> Option<(RuleGroup, Rule)> {
         (Flake8Pyi, "055") => (RuleGroup::Stable, rules::flake8_pyi::rules::UnnecessaryTypeUnion),
         (Flake8Pyi, "056") => (RuleGroup::Stable, rules::flake8_pyi::rules::UnsupportedMethodCallOnAll),
         (Flake8Pyi, "058") => (RuleGroup::Stable, rules::flake8_pyi::rules::GeneratorReturnFromIterMethod),
-        (Flake8Pyi, "057") => (RuleGroup::Preview, rules::flake8_pyi::rules::ByteStringUsage),
+        (Flake8Pyi, "057") => (RuleGroup::Stable, rules::flake8_pyi::rules::ByteStringUsage),
```
Comment (AlexWaygood) [inline crates/ruff_linter/src/codes.rs:782]: I think this one has been around for a little less than 90 days -- it was merged on May 29 and released on May 31. But it's also a very simple and uncontroversial rule (IMO), so I'm okay with stabilising it a couple of weeks before we normally would 😄 it's not in our versioning policy that we have to wait 90 days

**Review by AlexWaygood — APPROVED** [review]

**Comment by AlexWaygood** [comment]
I think the ecoystem report has gone haywire because you changed the base branch of the PR after it had been opened. Could you try rebasing it? Or maybe closing and reopening it?

File: crates/ruff_linter/src/codes.rs (hunk 1 above)
Comment (MichaReiser) [reply crates/ruff_linter/src/codes.rs:782]: Lol, that's correct. I only saw *May* and was like, that's a long time ago. I did do the proper math for the other rules.

**Comment by MichaReiser** [comment]
Hmm, I'm trying to reproduce the typeshed errors where the error message looks very off but without success so far

**Comment by AlexWaygood** [comment]
> Hmm, I'm trying to reproduce the typeshed errors where the error message looks very off but without success so far

If you can't repro them locally, it's probably another case of https://github.com/astral-sh/ruff/issues/11305

**Pull request merged**